### DISCUSSION TITLE
How to create typeorm relationships with next-auth user model

### DISCUSSION THREAD

**Discussion by kenperkins** [nextauthjs/next-auth#729] [Help]
**Your question**
I'm trying to create an extended schema of models outside of next-auth, but I want to be able to create associations to the user model. In the simplest case, I want to be able to use typeorm to query for models related to the next-auth user.

I could certainly do this without associations by just adding a field to my models of the same type as the user id, however, I'd love to know if I can do this more _correctly_ in the sense of a proper typeorm model.

I read through the extending or using my own models, and maybe that's the right way to do this, but it's opaque how to actually create my own model based on the limited documentation here: 

https://next-auth.js.org/tutorials/typeorm-custom-models#using-custom-models

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [ ] Found the example project helpful
* [x] Did not find the example project helpful

**Comment by RicardoAlmeida25** [comment]
I'm in the same situation, would love to know the answer to this.

**Comment by kamerondotcom** [comment]
I'm also having issues extending my user model following this documentation (https://next-auth.js.org/tutorials/typeorm-custom-models) with MongoDB.

**Comment by RicardoAlmeida25** [comment]
Not a solution to our issue here, but I tried prisma instead of typeorm and it's really easy to do it just so you know. To do this simply extend the base model as you want: https://next-auth.js.org/schemas/adapters.
